vscode/src/client/analyzerClient.ts

```diff
@@ -140,6 +141,15 @@ export class AnalyzerClient {
       new rpc.StreamMessageReader(this.kaiRpcServer.stdout),
       new rpc.StreamMessageWriter(this.kaiRpcServer.stdin),
     );
+
+    this.rpcConnection.onNotification("my_progress", (params) => {
+      if (params.kind === "SimpleChatMessage") {
+        this.fireSolutionStateChange("sent", params.value.message);
+      } else {
+        this.fireSolutionStateChange("sent", JSON.stringify(params));
+      }
+    });
```

Comment:
Could add a little bit of protection by only doing something if the solution state is already `"sent"`...
```suggestion
    this.rpcConnection.onNotification("my_progress", (params) => {
      if (this.getExtStateData().solutionState !== "sent") {
        return;
      }
      if (params.kind === "SimpleChatMessage") {
        this.fireSolutionStateChange("sent", params.value.message);
      } else {
        this.fireSolutionStateChange("sent", JSON.stringify(params));
      }
    });
```

Reply:
Perhaps I'm misunderstanding. Will this ever block a chat message from being sent?

Reply:
no. He was saying that his comment on your PR itself was non-blocking.

Reply:
I know this function will get some upgrades in future, so it is ok as-is at the moment.